workflow/scripts/plot-scatter.R

```diff
@@ -0,0 +1,30 @@
+log <- file(snakemake@log[[1]], open="wt")
+sink(log)
+sink(log, type="message")
+
+library("sleuth")
+library("tidyverse")
+library("ggpubr")
+
+so <- sleuth_load(snakemake@input[[1]])
+
+sample_matrix <- t(combn(so$sample_to_covariates$sample, 2))
+combinations <- as.numeric(row(sample_matrix))
+
+plot_list <- list()
+
+for(i in combinations) {
+  sample1 <- sample_matrix[i, 1]
+  sample2 <- sample_matrix[i, 2]
+  plot_list[[i]] <- plot_scatter(so, sample_x = sample1, sample_y = sample2, use_filtered = TRUE,
+               units = "est_counts", offset = 1, point_alpha = 0.2, xy_line = TRUE,
+               xy_line_color = "red", trans = "log", xlim = NULL, ylim = NULL)+ 
+    labs(x = sample1, y = sample2)
+}
+
+all_plots <- ggarrange(plotlist = plot_list)
+plot_figure <- annotate_figure(all_plots, 
+                top = text_grob("log transformed scatter plots of different samples",
+                                color = "black", face = "bold", size = 14)
+                )
```

Comment:
Here, I am not sure what we want. What do you think?

I see two options:

1. As you implemented it. This way, each combination is only plotted once and we avoid redundant plots.
2. Plot every combination of samples in a grid, similar to the arrangement [e.g. seen here](http://www.sthda.com/english/wiki/scatter-plot-matrices-r-base-graphs). This way, each combination is easier to find at a glance. To generate this, you could just iterate over all samples in two nested loops, i.e.:

```
for (i in so$sample_to_covariates$sample) {
  for (j in so$sample_to_covariates$sample) {
    add plot_scatter to list
 }
}
```

And then wrap the plots after every `length(so$sample_to_covariates$sample)`.

So, what do you think?

Reply:
I'm not too worried about nested for loops with a limited number of samples---but who knows, maybe people will throw in excessive amounts of samples in the future. So you may be right in being more systematic, here. So let's merge this plot as you implemented it now and have a look on some real data.